### PR TITLE
Create default implementation of get_columns_in_relation_sql

### LINKED ISSUE #51: Add support for spark
File "/usr/local/lib/python3.9/dist-packages/dbt/parser/manifest.py", line 526, in macro_depends_on
    possible_macro_calls = statically_extract_macro_calls(
  File "/usr/local/lib/python3.9/dist-packages/dbt/clients/jinja_static.py", line 43, in statically_extract_macro_calls
    ad_macro_calls = statically_parse_adapter_dispatch(
  File "/usr/local/lib/python3.9/dist-packages/dbt/clients/jinja_static.py", line 147, in statically_parse_adapter_dispatch
    macro = db_wrapper.dispatch(func_name, macro_namespace=macro_namespace).macro
  File "/usr/local/lib/python3.9/dist-packages/dbt/context/providers.py", line 186, in dispatch
    raise CompilationException(msg)
dbt.exceptions.CompilationException: Compilation Error
  In dispatch: No macro named 'get_columns_in_relation_sql' found
      Searched for: 'project.spark__get_columns_in_relation_sql', 'project.default__get_columns_in_relation_sql', 'audit_helper.spark__get_columns_in_relation_sql', 'audit_helper.default__get_columns_in_relation_sql'
11:23:16  Encountered an error:
Compilation Error
  In dispatch: No macro named 'get_columns_in_relation_sql' found
      Searched for: 'project.spark__get_columns_in_relation_sql', 'project.default__get_columns_in_relation_sql', 'audit_helper.spark__get_columns_in_relation_sql', 'audit_helper.default__get_columns_in_relation_sql'

### PR DESCRIPTION
**Pull request by joellabes**
## Description & motivation
Resolves #51 

To come: add some tests (admittedly these won't capture the spark issue because our matrix only covers BQ/SF/RS/PG, but at minimum we should be adding validation for those regardless).

## Future possibilities
If https://github.com/dbt-labs/dbt-core/issues/6449 exposes `ordinal_position` from each adapter in a future release of dbt-core, then a single `default__get_columns_in_relation_sql` (without adapter-specific overrides) may be sufficient.

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

**Review by dave-connors-3 — APPROVED** [review]
I was able to successfully run this on databricks! I threw the `compare_relation_columns` into a model, and saw the expected result:

image


Really simple workaround! Although the ordinal position is accurate in my example, since we don't guarantee it to be accurate at all, would it make sense to exclude the related columns from the output if you're using an "unsupported" adapter?

**Comment by joellabes** [comment]
@dave-connors-3 I did not have a fun afternoon, but I do now have tests up and going! 

> Although the ordinal position is accurate in my example, since we don't guarantee it to be accurate at all, would it make sense to exclude the related columns from the output if you're using an "unsupported" adapter?

Ironically, in the pursuit of testing I rediscovered one of my traumatic experiences from my Redshift days - you can't persist the results of an information_schema query (even as a view) – afaik it's because that data only lives on the leader node. Because of this, I wound up falling back to the default implementation. 

I think this is OK - the get_columns_in_relation macro also powers things like `dbt_utils.star` and I have never seen that behave non-deterministically. I don't know that consistent behaviour is strictly guaranteed anywhere, but:
- if dbt Core broke it I would claim that as a breaking change that shouldn't be allowed in a minor/patch release
- if a vendor broke it I would be very mad and I guess we'd just do something about it. 

lmk if you strongly disagree!

File: integration_tests/dbt_project.yml [hunk 1]
```diff
@@ -15,5 +15,8 @@ clean-targets:         # directories to be removed by `dbt clean`
     - "target"
     - "dbt_packages"
 
+# models:
+#   +materialized: table
+
```
Comment (dbeatty10) [inline integration_tests/dbt_project.yml:20]: Do these commented lines need to stay in the PR? Or were they included just while figuring this all out?

Comment (joellabes) [reply integration_tests/dbt_project.yml:20]: Good catch! they are indeed unnecessary

**Comment by dbeatty10** [comment]
> Are there outstanding blockers here?

@dave-connors-3 here's at least one blocker:

image

**Comment by joellabes** [comment]
once circle succeeds i will get this merged!
